ubuntu_image/helpers.py

```diff
@@ -214,9 +214,16 @@ def mkfs_ext4(img_file, contents_dir, label='writable',
     Ubuntu 16.04, which has e2fsprogs 1.42.X without the -d flag.  In
     that case, we have to sudo loop mount the ext4 file system and
     populate it that way.  Which sucks because sudo.
+    NOTE Unfortunately we cannot use fakeroot for classic, as that changes the
+    ownership of the files - critical for /home/*
     """
-    cmd = ('fakeroot-sysv mkfs.ext4 -L {} -O -metadata_csum -T default '
-           '-O uninit_bg {} -d {}').format(label, img_file, contents_dir)
```

Comment:
Ok, looking at the actual diff, the issue here is that fakeroot is being used to wrap *only* the mkfs.ext4 call.  When you are running as non-root, what is needed is to wrap the *entire* operation in a single fakeroot invocation - both the directory creation and the filesystem creation.  If you don't have root privileges, then not wrapping with fakeroot will also definitely give you wrong output.

If classic builds require real root to master the rootfs tree, then yes we should enforce this and avoid calling fakeroot.

I don't love that we are currently calling sudo unconditionally to invoke live-build; I think the code should check if we're already root and avoid calling sudo in that case.  But at any rate, once that filesystem is created you should be able to call mkfs.ext4 without *either* sudo or fakeroot since we are unconditionally using the mkfs.ext4 -d option.